docs/core/compatibility/7.0.md

```diff
@@ -45,9 +45,11 @@ If you're migrating an app to .NET 7, the breaking changes listed here might aff
 | [Collectible Assembly in non-collectible AssemblyLoadContext](core-libraries/7.0/collectible-assemblies.md) | ❌ | ✔️ | Preview 5 |
 | [Equals method behavior change for NaN](core-libraries/7.0/equals-nan.md) | ❌ | ✔️ | Preview 5 |
 | [Generic type constraint on PatternContext\<T>](core-libraries/7.0/patterncontext-generic-constraint.md) | ❌ | ❌ | Preview 3 |
+| [Legacy FileStream strategy removed](core-libraries/7.0/filestream-compat-switch.md) | ❌ | ✔️ | Preview 1 |
```

Comment:
I'm not really sure if this change is not binary/source compatible. My understanding based on https://stackoverflow.com/a/14908742/4231460 is that they are both, source and binary compatible.

cc @adamsitnik

Reply:
This is how I think of binary and source compatibility:

- Source compatible means when you recompile an app using the new SDK/library, there are no new errors/warnings.
- Binary compatible means you can run an app against the new runtime/library without recompilation or any behavior changes.

If there is a behavior difference between the legacy FileStream code and the new implementation, then I think you can say the change can break binary compatibility.

For the time fields on symlinks/targets, it seems even easier to say that it can break binary compatibility, because you might be expecting the time on the target to have changed, but it doesn't using the new library.

Reply:
> For the time fields on symlinks/targets, it seems even easier to say that it can break binary compatibility

I agree with this one.

> If there is a behavior difference between the legacy FileStream code and the new implementation, then I think you can say the change can break binary compatibility.

For this one, the breaking change is that the env. var./context switch no longer does anything; customers should remove their usage. FileStream code should work as if there wasn't a switch at all. 

So, I think both breaking changes are binary.